### DISCUSSION TITLE
How to rename the class member variables ?

### DISCUSSION THREAD

**Discussion by oviche** [NationalSecurityAgency/ghidra#5046] [Q&A]
Hi all,

I was working with Ghidra to analyze the dex file and I faced a problem in that I couldn't rename the class member variables. For example, in the below screenshot, I can't find a way to rename `this.f` can't be renamed.

![screenshot](https://user-images.githubusercontent.com/20806107/222013546-622b0bed-6d1f-4972-abc2-63e6056bbc5f.png)

So I wonder if this functionality is supported in Ghidra or not.

**Comment by astrelsky** [comment]
It looks like this only affects the Java decompiler. I'm not actually sure if you're supposed to be able to rename Java class fields or not but it doesn't look like rename variables either (which should work).

**Reply by oviche** [reply]
I find the rename option is only supported for the local variables in the functions but not for the class fields.

However, the only way I find to achieve that ( which is not a good idea) is to allocate a memory map and write the name of the variable there, then fix the reference so it points to the newly allocated memory map.
